rclcpp/include/rclcpp/subscription_traits.hpp

```diff
@@ -69,7 +73,19 @@ template<typename MessageT, typename Deleter>
 struct extract_message_type<std::unique_ptr<MessageT, Deleter>>: extract_message_type<MessageT>
 {};
 
-template<typename CallbackT>
+template<
+  typename CallbackT,
+  // Do not attempt if CallbackT is an integer (mistaken for depth)
+  typename = std::enable_if_t<!std::is_integral<CallbackT>::value>,
+  // Do not attempt if CallbackT is a QoS (mistaken for qos)
+  typename = std::enable_if_t<!std::is_same<
```

Comment:
I think that using here `std::is_base_of` will solve the last build problem in macOS.

Reply:
Yup, sounds right to me, thanks!